src/main/java/Entities/UserDataClasses/HideableUserDataClasses/Attributes.java

```diff
@@ -25,32 +28,59 @@ public Attributes(boolean hidden){
         this.hidden = hidden;
     }
 
-    public void setAttribute(Integer attributeKey, Integer valueKey){
+    // Methods
+    public void addAttribute(Integer attributeKey, Integer valueKey){
         // sets the attribute with key attributeKey (in AttributesDict) to the
         // value with valueKey (in AttributeValuesDict)
         this.data.put(attributeKey, new ArrayList<Object>());
         this.data.get(attributeKey).add(valueKey);
         this.data.get(attributeKey).add(true);
     }
-    public void setAttribute(Integer attributeKey, Integer valueKey, boolean hidden){
+    public void addAttribute(Integer attributeKey, Integer valueKey, boolean hidden){
 
         this.data.put(attributeKey, new ArrayList<Object>());
         this.data.get(attributeKey).add(valueKey);
         this.data.get(attributeKey).add(hidden);
     }
-
+    public void addAttributes(ArrayList<ArrayList<Integer>> attributeValuePairs){
+        // Precondition: Each sublist must be exactly of length 2, and the first
+        // number in each sublist should not be repeated
+        for(int i = 0; i < attributeValuePairs.size(); i++){
+            this.addAttribute(attributeValuePairs.get(i).get(0),
+                    attributeValuePairs.get(i).get(1));
+        }
+    }
     public void removeAttribute(Integer attributeKey){
         this.data.remove(attributeKey);
     }
-
+    public void removeAttributes(ArrayList<Integer> attributeKeys){
+        // Precondition: Each integer in attributeKeys should be unique
+        for(int i = 0; i < attributeKeys.size(); i++){
+            this.removeAttribute(attributeKeys.get(i));
+        }
+    }
+    public void changeAttribute(Integer attributeKey, Integer newValueKey){
+        // Precondition: Attribute with attributeKey has already been set to
+        // a value
+        this.data.get(attributeKey).clear();
+        this.data.get(attributeKey).add(newValueKey);
+        this.data.get(attributeKey).add(true);
+    }
     public void hideAttribute(Integer attributeKey){
         this.data.get(attributeKey).set(1, true);
     }
-
     public void showAttribute(Integer attributeKey){
         this.data.get(attributeKey).set(1, false);
     }
-    public void setVisibility(Integer attributeKey, boolean hidden){
+    public void setAttributeVisibility(Integer attributeKey, boolean hidden){
         this.data.get(attributeKey).set(1, hidden);
     }
+    public String getTypeAt(int attributeKey){
```

Comment:
I don't really get why you made all these "getTypeAt", "getValueAt" methods, can you not just use getData.get(attributekey)?

Reply:
No, getTypeAt is meant to return the type of attribute (eye color, height, etc.) as a string, whereas getData.get(attributeKey) would return the value associated with that attribute, or cause an error if that attribute has not been set yet. This method works even if the Attributes instance doesn't contain a value for attributeKey.